### PR TITLE
feat(mojaloop/#3636): v15.3.0 RC changes

### PR DESCRIPTION
**Pull request by oderayi**
v15.3.0 RC changes - [#3636](https://github.com/mojaloop/project/issues/3636)
- Implement retry logic for a number of flaky tests
- Switched a number of tests to HTTP endpoints as a temporary workarounds for TTK websocket issue

### CODE REVIEW

File: collections/hub/golden_path/feature_tests/transfer_negative_scenarios/fulfil-reserved-v1.0.json [hunk 1]
```diff
@@ -561,8 +565,69 @@
...
+                "",
+                "  for (let retryCount = 0 ; retryCount < maxRetryCount; retryCount++) {",
+                "    console.log(`retry-count=${retryCount+1}, max-count=${maxRetryCount}, max-wait=${maxRetryWait}`)",
+                "    ",
+                "    const isOk = _callback?.body?.transferState === expectedTransferState;",
+                "    ",
+                "    if (isOk) {",
+                "      console.log(`Everything is OK, let's continue`)",
+                "      environment.transferResponse = _response",
+                "      environment.transferCallback = _callback",
```
Comment (vijayg10) [inline collections/hub/golden_path/feature_tests/transfer_negative_scenarios/fulfil-reserved-v1.0.json:594]: If you don't need to access these variables in following requests in the test case, you can use requestVariables also

File: collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json [hunk 2]
```diff
@@ -436,16 +436,18 @@
           "scripts": {
             "preRequest": {
               "exec": [
-                "if (environment.ENABLE_WS_ASSERTIONS===true) {",
-                "  await websocket.connect(environment.PAYEEFSP_SDK_TESTAPI_WS_URL + '/requests/{$inputs.toIdValue}', 'payeeRequest')",
-                "}"
+                "// if (environment.ENABLE_WS_ASSERTIONS===true) {",
+                "//   await websocket.connect(environment.PAYEEFSP_SDK_TESTAPI_WS_URL + '/requests/{$inputs.toIdValue}', 'payeeRequest')",
+                "// }"
```
Comment (vijayg10) [inline collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json:441]: Can you add TODO comments mentioning why you commented this code?

File: collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json [hunk 3]
```diff
@@ -436,16 +436,18 @@
...
             },
             "postRequest": {
               "exec": [
-                "if (environment.ENABLE_WS_ASSERTIONS===true) {",
-                "  environment.payeeRequest = await websocket.getMessage('payeeRequest', environment.WS_ASSERTION_TIMEOUT)",
-                "}"
+                "// if (environment.ENABLE_WS_ASSERTIONS===true) {",
+                "//   environment.payeeRequest = await websocket.getMessage('payeeRequest', environment.WS_ASSERTION_TIMEOUT)",
+                "// }",
+                "",
```
Comment (vijayg10) [inline collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json:449]: Can you add TODO comments mentioning why you commented this code?

File: collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json [hunk 4]
```diff
@@ -636,16 +638,18 @@
           "scripts": {
             "preRequest": {
               "exec": [
-                "if (environment.ENABLE_WS_ASSERTIONS===true) {",
-                "  await websocket.connect(environment.PAYEEFSP_SDK_TESTAPI_WS_URL + '/requests/{$request.body.quoteId}', 'payeeRequest')",
-                "}"
+                "// if (environment.ENABLE_WS_ASSERTIONS===true) {",
+                "//   await websocket.connect(environment.PAYEEFSP_SDK_TESTAPI_WS_URL + '/requests/{$request.body.quoteId}', 'payeeRequest')",
+                "// }"
```
Comment (vijayg10) [inline collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json:643]: Can you add TODO comments mentioning why you commented this code?

File: collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json [hunk 5]
```diff
@@ -636,16 +638,18 @@
...
               ]
             },
             "postRequest": {
               "exec": [
-                "if (environment.ENABLE_WS_ASSERTIONS===true) {",
-                "  environment.payeeRequest = await websocket.getMessage('payeeRequest', environment.WS_ASSERTION_TIMEOUT)",
-                "}"
+                "// if (environment.ENABLE_WS_ASSERTIONS===true) {",
+                "//   environment.payeeRequest = await websocket.getMessage('payeeRequest', environment.WS_ASSERTION_TIMEOUT)",
+                "// }",
```
Comment (vijayg10) [inline collections/hub/golden_path/feature_tests/backward_compatibility/fspiop_protocol_validation.json:650]: Can you add TODO comments mentioning why you commented this code?

File: collections/hub/other_tests/bulk_transfers/bulk-invalid-timestamp.json [hunk 6]
```diff
@@ -338,69 +343,122 @@
...
+                "",
+                "  for (let retryCount = 0 ; retryCount < maxRetryCount; retryCount++) {",
+                "    console.log(`retry-count=${retryCount+1}, max-count=${maxRetryCount}, max-wait=${maxRetryWait}`)",
+                "    ",
+                "    const isOk = _callback?.body?.bulkTransferState === expectedBulkTransferState;",
+                "    ",
+                "    if (isOk) {",
+                "      console.log(`Everything is OK, let's continue`)",
+                "      environment.bulkTransferResponse = _response",
+                "      environment.bulkTransferCallback = _callback",
```
Comment (vijayg10) [inline collections/hub/other_tests/bulk_transfers/bulk-invalid-timestamp.json:421]: If you don't need to access these variables in other requests, you can use requestVariables.

File: collections/hub/other_tests/bulk_transfers/bulk-invalid-timestamp.json [hunk 7]
```diff
@@ -427,18 +485,64 @@
...
+                "async function main() {",
+                "  let _response = response",
+                "",
+                "  for (let retryCount = 0 ; retryCount < maxRetryCount; retryCount++) {",
+                "    console.log(`retry-count=${retryCount+1}, max-count=${maxRetryCount}, max-wait=${maxRetryWait}`)",
+                "    ",
+                "    const isOk = _response.code === expectedStatusCode;",
+                "    ",
+                "    if (isOk) {",
+                "      environment.participantsPositionResponse = _response",
```
Comment (vijayg10) [inline collections/hub/other_tests/bulk_transfers/bulk-invalid-timestamp.json:520]: If you don't need to access these variables in other requests, you can use requestVariables.

File: collections/hub/other_tests/bulk_transfers/bulk-invalid-timestamp.json [hunk 8]
```diff
@@ -464,17 +568,63 @@
...
```
Comment (vijayg10) [inline collections/hub/other_tests/bulk_transfers/bulk-invalid-timestamp.json:604]: If you don't need to access these variables in other requests, you can use requestVariables.

File: collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json [hunk 9]
```diff
@@ -301,18 +302,63 @@
...
+                " async function main() {",
+                "   let _response = response",
+                " ",
+                "   for (let retryCount = 0 ; retryCount < maxRetryCount; retryCount++) {",
+                "     console.log(`retry-count=${retryCount+1}, max-count=${maxRetryCount}, max-wait=${maxRetryWait}`)",
+                "     ",
+                "     const isOk = _response.code === expectedStatusCode && _response.body[0].value === (+environment.payerfspPositionBeforePrepare)",
+                "     ",
+                "     if (isOk) {",
+                "       environment.participantsPositionResponse = _response",
```
Comment (vijayg10) [inline collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json:336]: If you don't need to access these variables in other requests, you can use requestVariables.

File: collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json [hunk 10]
```diff
@@ -338,17 +384,62 @@
...
+                " async function main() {",
+                "   let _response = response",
+                " ",
+                "   for (let retryCount = 0 ; retryCount < maxRetryCount; retryCount++) {",
+                "     console.log(`retry-count=${retryCount+1}, max-count=${maxRetryCount}, max-wait=${maxRetryWait}`)",
+                "     ",
+                "     const isOk = _response.code === expectedStatusCode && _response.body[0].value === (+environment.payeefspPositionBeforePrepare)",
+                "     ",
+                "     if (isOk) {",
+                "       environment.participantsPositionResponse = _response",
```
Comment (vijayg10) [inline collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json:419]: If you don't need to access these variables in other requests, you can use requestVariables.

File: collections/hub/provisioning/CGS_Specific/adjust-participants-limits.json [hunk 11]
```diff
@@ -32,12 +32,60 @@
...
+                "async function main() {",
+                "  let statusCode = response.code",
+                "",
+                "  for (let retryCount = 0 ; retryCount < maxRetryCount; retryCount++) {",
+                "    console.log(`retry-count=${retryCount+1}, max-count=${maxRetryCount}, max-wait=${maxRetryWait}`)",
+                "    ",
+                "    const isOk = statusCode === expectedStatusCode;",
+                "    ",
+                "    if (isOk) {",
+                "      environment.participantLimitsResponseCode = statusCode",
```
Comment (vijayg10) [inline collections/hub/provisioning/CGS_Specific/adjust-participants-limits.json:62]: If you don't need to access these variables in other requests, you can use requestVariables.

File: collections/hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json [hunk 12]
```diff
@@ -5,10 +5,7 @@
       "id": 1,
       "name": "TC-BHP1. When there are 4 transfers in the bulk request 2 of which that have the same receiver fsp id and the others has a different receiver fsp id and acceptParty is true for all the transfers, Then the PUT response should have details of all the 4 transfers.",
       "fileInfo": {
-        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json",
-        "labels": [
-          "basic"
-        ]
+        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json"
```
Comment (vijayg10) [inline collections/hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json:8]: Why labels are removed?

Comment (oderayi) [reply collections/hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json:8]: @vijayg10 The labels are removed by TTK on export.

Comment (oderayi) [inline collections/hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json:8]:
```suggestion
       "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json",
        "labels": [
          "basic"
        ]
```

File: collections/hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json [hunk 13]
```diff
@@ -596,10 +593,7 @@
       "id": 2,
       "name": "TC-BHP3. When the incoming request has a format issue that fails schema validation, Then the return code should be 400 instead of 202",
       "fileInfo": {
-        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json",
-        "labels": [
-          "basic"
-        ]
+        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json"
```
Comment (oderayi) [inline collections/hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json:596]:
```suggestion
        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-happy-path.json",
        "labels": [
          "basic"
        ]
```

File: collections/hub/sdk_scheme_adapter/bulk/basic/bulk-high-volume_dynamic.json [hunk 14]
```diff
@@ -405,10 +405,7 @@
         }
       ],
       "fileInfo": {
-        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-high-volume_dynamic.json",
-        "labels": [
-          "basic"
-        ]
+        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-high-volume_dynamic.json"
```
Comment (oderayi) [inline collections/hub/sdk_scheme_adapter/bulk/basic/bulk-high-volume_dynamic.json:408]:
```suggestion
        "path": "hub/sdk_scheme_adapter/bulk/basic/bulk-high-volume_dynamic.json",
        "labels": [
          "basic"
        ]
```

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**